src/coreclr/jit/runtimelookup.cpp

```diff
@@ -141,17 +143,22 @@ PhaseStatus Compiler::fgExpandRuntimeLookups()
                     asgStmt->SetDebugInfo(debugInfo);
                     gtSetStmtInfo(asgStmt);
                     fgSetStmtSeq(asgStmt);
+                    gtUpdateStmtSideEffects(asgStmt);
                     return gtNewLclvNode(tmpNum, expr->TypeGet());
                 };
 
-                // if sigTree was not a constant e.g. COMMA(..., CNS)) - spill it
-                if (!sigTree->IsCnsIntOrI())
+                GenTree* ctxTree = call->gtArgs.GetArgByIndex(0)->GetNode();
+                GenTree* sigNode = call->gtArgs.GetArgByIndex(1)->GetNode();
+                if (!ctxTree->OperIs(GT_LCL_VAR))
+                {
+                    ctxTree = spillExpr(ctxTree);
+                }
+                if (!sigNode->OperIs(GT_LCL_VAR, GT_CNS_INT))
                 {
-                    spillExpr(sigTree);
+                    sigNode = spillExpr(sigNode);
                 }
```

Comment:
Don't think this should be needed anymore with `gtSplitTree`

Reply:
@jakobbotsch afair, when I enable CSE for constants I might get here `COMMA(ASG)`

Reply:
`gtSplitTree` will ensure that the arguments of the call are split out if they have any sort of side effects -- it should be ok to always clone any operand to the call here.

Reply:
@jakobbotsch so are you saying that I should never hit COMMA here? because if I do I definitely need to spill it

Reply:
Yes, we shouldn't see COMMA here

Reply:
Fixed with your suggesion